src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,32 @@
+import Popup from "./Popup";
+
+class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._deleteButton = this._popupElement.querySelector(
+      ".modal__delete-button"
+    );
+  }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  _handleSubmit = (event) => {
+    event.preventDefault();
+    this._handleFormSubmit();
+  };
+
+  close() {
+    super.close();
+    this._popupForm.addEventListener("submit", this._handleSubmit);
```

Comment:
Please, delete method `close` from `PopupWithConfirmation`. You don't need it. And you add the listener istead of removing. 


But you don't need to remove the listener because you update it in `setSubmitAction`